From: Vivek Bhagat <vivek.bhagat89-Re5JQEeQqe8AvxtiuMwx3w@public.gmane.org>
To: netdev-u79uwXL29TY76Z2rM5mHXA@public.gmane.org,
	linux-usb-u79uwXL29TY76Z2rM5mHXA@public.gmane.org
Subject: [Patch] ax88179_178a: add reset functionality in reset_resume
Date: Sat, 20 Jun 2015 23:24:34 +0530	[thread overview]
Message-ID: <CA+vmRF4FpAhc-8QQhSY+mZg-a2R3aP5Qy1FBgv0VwNtE3iLkBA@mail.gmail.com> (raw)

[-- Attachment #1: Type: text/plain, Size: 295 bytes --]

 Dear All,

Attached patch fix iperf connection problem after reset resume of
ethernet to usb dongle.

Without reset functionality, i see ping works after reset resume but
iperf connection fails with wrong checksum error message shown by
tcpdump.

Attached patch fix above issue.

Thanks,
Vivek

[-- Attachment #2: 0001-ax88179_178a-add-reset-function-in-reset_resume.patch --]
[-- Type: application/octet-stream, Size: 1661 bytes --]

From d178065c9e3cfa8a45ef537fae7412775339beb0 Mon Sep 17 00:00:00 2001
From: Vivek Kumar Bhagat <vivek.bhagat@samsung.com>
Date: Thu, 11 Jun 2015 07:23:46 -0700
Subject: [PATCH] ax88179_178a: add reset function in reset_resume

without reset functionality in reset_resume, iperf connection
does not establish after suspend/resume however ping works at
the same time.

reset function inside reset_resume solves above bug. We have verified
it on ASIX based ST Lab, Cadyce dongle.

Signed-off-by: Vivek Kumar Bhagat <vivek.bhagat@samsung.com>
Signed-off-by: Praveen Kumar <praveen.k09@samsung.com>

---
 drivers/net/usb/ax88179_178a.c |   14 +++++++++++++-
 1 file changed, 13 insertions(+), 1 deletion(-)

diff --git a/drivers/net/usb/ax88179_178a.c b/drivers/net/usb/ax88179_178a.c
index e6338c1..00928c0 100644
--- a/drivers/net/usb/ax88179_178a.c
+++ b/drivers/net/usb/ax88179_178a.c
@@ -1630,6 +1630,18 @@ static int ax88179_stop(struct usbnet *dev)
 	return 0;
 }
 
+static int ax88179_reset_resume(struct usb_interface *intf)
+{
+	struct usbnet *dev = usb_get_intfdata(intf);
+	int ret;
+
+	ret = ax88179_reset(dev);
+	if (ret < 0)
+		return ret;
+
+	return  ax88179_resume(intf);
+}
+
 static const struct driver_info ax88179_info = {
 	.description = "ASIX AX88179 USB 3.0 Gigabit Ethernet",
 	.bind = ax88179_bind,
@@ -1744,7 +1756,7 @@ static struct usb_driver ax88179_178a_driver = {
 	.probe =	usbnet_probe,
 	.suspend =	ax88179_suspend,
 	.resume =	ax88179_resume,
-	.reset_resume =	ax88179_resume,
+	.reset_resume =	ax88179_reset_resume,
 	.disconnect =	usbnet_disconnect,
 	.supports_autosuspend = 1,
 	.disable_hub_initiated_lpm = 1,
-- 
1.7.9.5

